modules/ui/map3d_viewer.js

```diff
@@ -15,10 +15,11 @@ export function uiMap3dViewer(context) {
   const l10n = context.systems.l10n;
   const map = context.systems.map;
   const map3d = context.systems.map3d;
+  const urlhash = context.systems.urlhash;
 
   function render(selection) {
     let wrap = d3_select(null);
-    let _isHidden = true; // start out hidden
+    let _isHidden = !context.systems.urlhash?.getParam('map3d'); // depends on URL hash
```

Comment:
```suggestion
    let _isHidden = !urlhash.getParam('map3d'); // depends on URL hash
```

some minor nitpicks 
- `urlhash` variable is already in scope here, so you can just refer to it rather than chaining to it from context.   
- Also the optional chaining is not needed, because if the url hash system doesn't exist, an exception would be thrown long before we get here.  By using `.?`, it jumped out at me, making me wonder what's going on.  "how can that go missing?".
- To ensure that that happens, make sure that 'urlhash' is added to the dependencies array in `Map3dSystem` - that is an init-time assert that will cause `Map3dSystem.initAsync` to throw if anything it depends on is missing.

Reply:
Ah right this is a better way to ensure that we don't fire too early!

Reply:
Awesome, thanks @bhousel